Design.md

```diff
@@ -0,0 +1,55 @@
+# Design doc for job-runner
+
+Job-runner provides a gRCP API to execute arbitrary commands on a Linux host.
+
+## Runner
+
+The jobs themselves are normal child processes of the runner. The runner configures CGroups and Linux namespaces for children to limit resource usage and isolate the them from the other processes. It also handles output recording and forwarding. All data is stored in-memory. In case of the service process termination, all jobs and their status are lost.
```

Comment:
please provide more detail about cgroups and namespaces:
- cgroups version
- specific cgroups controllers used and what limits are set
- is there one shared cgroup or one per child?
- which namespaces are configured

Reply:
I will provide full details in the next revision. However, currently known are:
* cgroups v1  is used
* the actual limits will be decided later, they shouldn't matter much as I'm going to hard-code those for simplicity
* one cgroup per child
* namespaces: pid, mount and network
  * network namespace will additionally require a VETH tunnel to do any networking